### PR TITLE
Charlesmchen/read receipt preferences

### PR DESCRIPTION
**Pull request by charlesmchen**
* Rename app preferences class.
* Add read receipts preference.
* New users opt-out of read receipts; legacy users opt-in.
* Add setting for read receipts in app settings.

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/environment/Environment.h [hunk 1]
```diff
@@ -2,10 +2,10 @@
 //  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
 //
 
-#import <Foundation/Foundation.h>
-#import "PropertyListPreferences.h"
+#import "OWSPreferences.h"
 #import "TSGroupModel.h"
 #import "TSStorageHeaders.h"
+#import <Foundation/Foundation.h>
```
Comment (michaelkirk) [inline Signal/src/environment/Environment.h]: this is in our pch, you can remove it.

File: Signal/src/environment/OWSPreferences.h [hunk 2]
```diff
@@ -52,13 +52,15 @@ extern NSString *const PropertyListPreferencesKeyEnableDebugLog;
...
 
-
 - (BOOL)hasDeclinedNoContactsView;
 - (void)setHasDeclinedNoContactsView:(BOOL)value;
 
 - (void)setIOSUpgradeNagVersion:(NSString *)value;
 - (nullable NSString *)iOSUpgradeNagVersion;
 
+- (BOOL)areReadReceiptsEnabled;
+- (void)setAreReadReceiptsEnabled:(BOOL)value;
```
Comment (michaelkirk) [inline Signal/src/environment/OWSPreferences.h]: TIOLI: are you opposed to using properties for these?

File: Signal/src/environment/OWSPreferences.m [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+{
+    ows_require(key != nil);
+
+    [TSStorageManager.sharedManager setObject:value forKey:key inCollection:OWSPreferencesSignalDatabaseCollection];
+}
+
+#pragma mark - Specific Preferences
+
+- (NSTimeInterval)getCachedOrDefaultDesiredBufferDepth
+{
```
Comment (michaelkirk) [inline Signal/src/environment/OWSPreferences.m]: Pretty sure this is redphone and can go away.

File: Signal/src/environment/OWSPreferences.m [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+    NSNumber *preference = [self tryGetValueForKey:OWSPreferencesKeyHasArchivedAMessage];
+    if (preference) {
+        return [preference boolValue];
+    } else {
+        return NO;
+    }
+}
+
+- (BOOL)hasRegisteredVOIPPush
```
Comment (michaelkirk) [inline Signal/src/environment/OWSPreferences.m]: doesn't look like we're using this either.

File: Signal/src/environment/OWSPreferences.m [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+        return YES;
+    }
+}
+
+- (void)setScreenSecurity:(BOOL)flag
+{
+    [self setValueForKey:OWSPreferencesKeyScreenSecurity toValue:@(flag)];
+}
+
+- (void)setHasRegisteredVOIPPush:(BOOL)enabled
```
Comment (michaelkirk) [inline Signal/src/environment/OWSPreferences.m]: Although we're setting this, we're never reading it, so it should go away.

**Comment by charlesmchen** [comment]
@michaelkirk Whoops I merged without addressing your comments. I'll do so now.

File: Signal/src/environment/Environment.h (hunk 1 above)
Comment (charlesmchen) [reply Signal/src/environment/Environment.h]: Removed here and throughout the codebase.

File: Signal/src/environment/OWSPreferences.h (hunk 2 above)
Comment (charlesmchen) [reply Signal/src/environment/OWSPreferences.h]: Nice idea. I'll try to do it the next time I touch this file.

File: Signal/src/environment/OWSPreferences.m (hunk 3 above)
Comment (charlesmchen) [reply Signal/src/environment/OWSPreferences.m]: Removed.

File: Signal/src/environment/OWSPreferences.m (hunk 4 above)
Comment (charlesmchen) [reply Signal/src/environment/OWSPreferences.m]: Done.

File: Signal/src/environment/OWSPreferences.m (hunk 5 above)
Comment (charlesmchen) [reply Signal/src/environment/OWSPreferences.m]: Done.

**Comment by charlesmchen** [comment]
@michaelkirk Addressed these here: https://github.com/WhisperSystems/Signal-iOS/commit/3d3f3bb590a73806055dc6114418a448d9209e38

**Pull request merged**